backtrack/permutation.go

```diff
@@ -20,22 +20,22 @@ func Permutation(nums []int) [][]int {
 	perm := make([]int, n)
 	vis := make([]bool, n)
 	var permutations [][]int
-	var backtrackPerm func(int)
-	backtrackPerm = func(depth int) {
-		if depth == n {
-			permutations = append(permutations, append([]int{}, perm...))
-			return
-		}
-		for i := range nums {
-			if vis[i] || (i > 0 && !vis[i-1] && nums[i] == nums[i-1]) {
-				continue
-			} // get first element that never used/visited
-			perm[depth] = nums[i]
-			vis[i] = true
-			backtrackPerm(depth + 1)
-			vis[i] = false
-		}
-	}
-	backtrackPerm(0)
+	backtrackPerm(&permutations, nums, 0, n, perm, vis)
 	return permutations
 }
+
+func backtrackPerm(permutations *[][]int, nums []int, depth int, n int, perm []int, vis []bool) {
+	if depth == n {
```

Comment:
If you're not passing the `nums` like this `nums[a:b]` for some a, b then yes it can be done that way, and the Len result is also constant so it doesn't change much.